### ISSUE TITLE
Textarea Editor behaves in consistently between edit and cancel

### ISSUE THREAD

**Issue by genyded** [olifolkerd/tabulator#2817]
**Describe the bug**
If editing a textarea with a custom formatter, if you edit the data then blur the cell, the editor resizes (desired in our case) and displays a single line of data. However if you cancel the edit the size stays as multi-line. It should either be one or the other and we would prefer single line even on cancel in this case.

**Tabulator Info**
- 4.6.3

**Working Example**
https://jsfiddle.net/nsk3x1p9/7/

**To Reproduce**
A step by step guide to recreate the issue in your JS Fiddle or Codepen:
1. Go to https://jsfiddle.net/nsk3x1p9/7/
2. Edit the 'Test One' column then blur that cell. You should see the cell height go to single line with an ellipsis upon blur
3. Prepare to edit the 'Test One' column again by focusing on it, but this time 'cancel' the edit by either not changing anything and bluring, or hitting ESC
4. You should see the cell height stays at the editor height instead of going back to single line like an actual edit does

**Expected behavior**
Both edit and clear should behave the same when bluring the cell (either both keep Editor height or both keep single line height). We would prefer both keep single line height.

**Desktop (please complete the following information):**
 - OS: Win 10
 - Browser Any but using Chrome

**Comment by genyded** [comment]
Anyone home?

**Comment by olifolkerd** [comment]
Hey @genyded 

Thanks for your JS Fiddle, that was very helpful. That is an interesting approach you have there with a variable height editor and a fixed height cell. a usage case i havnt seen before, which is where the issue arrises.

Is there a particular reason you are formatting the data in this way? it seems a bit jarring to have the editor experience be so different from the formatted data.

That being said it is definitely a bug so i will look at getting a fix in for that in the next release.

On a side note, please dont chase me for a response, especially after only one day. I provide this library for free, built on my own time. I cannot always be available to respond to people instantly. I get back to users as often as i can when i have the time.

Cheers

Oli :)

**Comment by genyded** [comment]
Hi Oli,

The reason is that we actually want the textarea to show single line (with an ellipsis if needed) when not being edited, nut lager when being edited. That is because 100 rows of large textareas eats up a lot of real estate and in our case the textarea content is only of real value when being edited. It's more important when not editing for them to see more cells from other rows than the textarea content for a few. However we need to show a portion of it when not being edited so that the users can tell if it has a value or not.(and in out case whether or not it's required if it has no value yet). Hope that makes at least some sense, Our use cases are quite complex.

I think we can find a suitable work around for now and may submit a PR if we come up with an actual fix. Still working on finalizing some other things on the overall data/grid first. 

RE: Chasing you... We realize this is a side project for you, but my tool choices also have to me our deadlines or it's my behind. We LOVE Tabulator so far and will try to be a bit better about not nagging, but we may get a bit of pressure on our end to show some progress or find another tool. So I am just trying to do what is needed to juggle out workloads at any point in time.

**Comment by olifolkerd** [comment]
Hey @genyded 

Thanks for your kind words, it is always great to hear that people are enjoying Tabulator.

That sounds sensible, it should be a quick fix. essentially the row height adjustment is currently only triggered on data change to prevent unnecessary redraws, it should be a quick fix to add this in if there is a combination of variable height editor and fixed height cell.

As a work around you could look at using the ***cellEditCancelled*** callback to trigger the ***normalizeHeight*** function on the row component, which should prevent the issue arrising.

```js
{title:"Test One", field:"test", editor:"textarea", formatter: rph, variableHeight: false, minWidth: '115', cellEditCancelled:function(cell){
    //cell - cell component

    cell.getRow().normalizeHeight();
}}
```
Thanks for your explanation. Tabulator is very much a passion of mine, not just a side project. Tabulator has 100,000's of users and some times people can be a bit overly demanding so i just prefer to make sure people are aware that i cant offer real time support, to manage people expectations. Im very happy to answer any and all questions, i just cant always guarantee it will be quick.

Cheers

Oli  :)

**Comment by genyded** [comment]
Thanks Oli! That works great!

However there is a similar issue if you click the cell instead of using keys to navigate to it. If you use keys, the textarea expands when it gets the focus. However if you click on it, it's single line with a scroll bar. Shouldn't it expand when clicked on too? Let me know if you want me to open another github issue for that or have any thoughts on a quick fix,

**Comment by rathboma** [comment]
@olifolkerd++ this works exactly right.

Seriously, I've never found a question unanswered by you so far.

**Comment by olifolkerd** [comment]
Hey @rathboma 

Thanks for the kind words! much appreciated :D

**Comment by olifolkerd** [comment]
@rathboma I love your beekeeper studio app.

If you are up for a chat, i would love to talk over some ideas for some other Tabulator features you could integrate for added functionality.

Oli  :)

**Comment by rathboma** [comment]
Yeah that'd be great. Wanna email me? First name dot last name at Gmail?

**Comment by olifolkerd** [comment]
Great, I will drop you a message tommorrow 😁

**Comment by olifolkerd** [comment]
Hey All,

I have just pushed a fix for this to the 4.8 branch which will be released next weekend.

Cheers

Oli :)
